electrolytes/solvation_shell_extract.py

```diff
@@ -0,0 +1,181 @@
+        dim = ts.triclinic_dimensions
+        box_center = np.sum(dim, axis=0) / 2
+        universe.atoms.translate(box_center - solu_center)
+
+        universe.atoms.unwrap()
+
+    solvent = universe.atoms - solu
+
+    solv_anal = Solute.from_atoms(solu, {"solvent": solvent}, solute_name=solute_atom)
+
+    # Identify the cutoff for the first solvation shell, based on the MD trajectory
+    logging.info("Running solvation analysis")
+    solv_anal.run()
+
+    # Plot the RDF
+    solv_anal.plot_solvation_radius("solute", "solvent")
+    plt.savefig(os.path.join(save_dir, system_name, solute_atom, "solvation_rdf.png"))
+
+    # There's probably a much faster way to do this
+    # But for now, we're prototyping, so slow is okay
+    shells = dict()
+    for j in solv_anal.speciation.speciation_fraction["solvent"]:
+        shells[j] = solv_anal.speciation.get_shells({"solvent": j})
+
+    # Now let's try getting the most diverse structures for each particular coordination number
+    # This is also a bit slow, particularly for the more common and/or larger solvent shells
+    for c in range(min_coord, max_coord + 1):
+        logging.info(f"Processing shells with coordination number {c}")
+        os.makedirs(
+            os.path.join(save_dir, system_name, solute_atom, f"coord={c}"),
+            exist_ok=True,
+        )
+        shell_species = list()
+        shell_positions = list()
+        for index, _ in tqdm(shells[c].iterrows()):
+            ts = universe.trajectory[index[0]]
+            universe.atoms.unwrap()
+            shell = solv_anal.solvation_data.xs(
+                (index[0], index[1]), level=(FRAME, SOLUTE_IX)
+            )
+            shell = solv_anal._df_to_atom_group(shell, solute_index=index[1])
+            shell = shell.copy()
+            if len(shell.atoms.elements) > len(shell_species):
+                shell_species = shell.atoms.elements
+
+            shell_positions.append(shell.atoms.positions)
+
+        by_num_atoms = defaultdict(list)
+        for sps in shell_positions:
+            by_num_atoms[len(sps)].append(sps)
+
+        selections_by_num_atoms = {
+            k: filter_by_rmsd(v, top_n) for k, v in by_num_atoms.items()
+        }
+
+        for k, v in selections_by_num_atoms.items():
+            for ii, positions in enumerate(v):
+                if positions.shape[0] == shell_species.shape[0]:
+                    # Wrap positions
+                    displacements = (
+                        positions[:, np.newaxis, :] - positions[np.newaxis, :, :]
+                    )
+                    idx = np.where(displacements > lattices / 2)[0]
+                    dim = np.where(displacements > lattices / 2)[2]
+                    if idx.shape[0] > 0:
+                        positions[idx, dim] -= lattices[0, 0, dim]
+                    # Save shell as xyz file
+                    mol = Molecule(shell_species, positions, charge=-1)
```

Comment:
Why is the charge always -1?

Reply:
@espottesmith this was something in your original notebooks. Can you weigh in here?

Reply:
Charge should not always be -1. My original scripts were for specific target systems.

Reply:
The charge should vary based on what the charges of the individual components are.